src/test/resources/regressions/features/domains/implements-interface-simple02.gobra

```diff
@@ -0,0 +1,53 @@
+// Any copyright is dedicated to the Public Domain.
+// http://creativecommons.org/publicdomain/zero/1.0/
+
+// this testcase checks whether a domain can implement a non-empty interface
+
+package ImplementsInterfaceSimple01
+
+ghost type intPair domain {
+    func fst(intPair) int
+    func snd(intPair) int
+    func pair(int, int) intPair
+
+    axiom {
+      forall p intPair :: {fst(p)}{snd(p)} p == pair(fst(p),snd(p))
+    } // pair
+
+    axiom {
+      forall l, r int :: {pair(l,r)} l == fst(pair(l,r)) && r == snd(pair(l,r))
+    }
+}
+
+type Equality interface {
+    ghost
+    pure hash() int
+
+    ghost
+    isEqual(other Equality) bool
+}
```

Comment:
do you think it would be worth adding a test where the interface has a non-ghost method? (my guess is no, as this should be already tested somewhere else)

Reply:
I've added two more test cases with an `Equality` interface specifying that both functions are actual. One test case checks that you cannot implement this interface by implementing ghost functions (ghostness mismatch for the function) and the second test case checks that implementing actual functions for a receiver of domain fails (actual function cannot have a ghost receiver)